src/main/java/net/dries007/tfc/compat/crafttweaker/CTItemRegistry.java

```diff
@@ -0,0 +1,235 @@
+        IIngredient inputIngredient = CTHelper.getInternalIngredient(input);
+        Size size = Size.valueOf(inputSize.toUpperCase());
+        Weight weight = Weight.valueOf(inputWeight.toUpperCase());
+        if (CapabilityItemSize.CUSTOM_ITEMS.get(inputIngredient) != null)
+        {
+            throw new IllegalStateException("Input registered more than once!");
+        }
+        else
+        {
+            CraftTweakerAPI.apply(new IAction()
+            {
+                @SuppressWarnings("unchecked")
+                @Override
+                public void apply()
+                {
+                    CapabilityItemSize.CUSTOM_ITEMS.put(inputIngredient, () -> new ItemSizeHandler(size, weight, true));
+                }
+
+                @Override
+                public String describe()
+                {
+                    return "Registered size and weight for " + input.toCommandString();
+                }
+            });
+        }
+    }
+
+    @ZenMethod
+    public static void registerItemHeat(crafttweaker.api.item.IIngredient input, float heatCapacity, float meltTemp, boolean forgeable)
+    {
+        if (input == null) throw new IllegalArgumentException("Input not allowed to be empty!");
+        if (input instanceof ILiquidStack)
+            throw new IllegalArgumentException("There is a fluid where it's supposed to be an item!");
+        if (heatCapacity <= 0 || meltTemp <= 0)
+            throw new IllegalArgumentException("Heat capacity and melt temp must be higher than 0!");
+        IIngredient inputIngredient = CTHelper.getInternalIngredient(input);
+        if (CapabilityItemHeat.CUSTOM_ITEMS.get(inputIngredient) != null || CapabilityForgeable.CUSTOM_ITEMS.get(inputIngredient) != null)
+        {
+            throw new IllegalStateException("Input already registered in forge/heat capability!");
+        }
+        else
+        {
+            CraftTweakerAPI.apply(new IAction()
+            {
+                @SuppressWarnings("unchecked")
+                @Override
+                public void apply()
+                {
+                    if (forgeable)
+                    {
+                        CapabilityForgeable.CUSTOM_ITEMS.put(inputIngredient, () -> new ForgeableHandler(null, heatCapacity, meltTemp));
+                    }
+                    else
+                    {
+                        CapabilityItemHeat.CUSTOM_ITEMS.put(inputIngredient, () -> new ItemHeatHandler(null, heatCapacity, meltTemp));
+                    }
+                }
+
+                @Override
+                public String describe()
+                {
+                    return "Registered heat capacity for " + input.toCommandString();
+                }
+            });
+        }
+    }
+
+    @ZenMethod
+    public static void registerHeatRecipe(String registryName, IItemStack input, IItemStack output, float transformTemp, float maxTemp)
```

Comment:
My suggestion for this:
 - Move it to it's own class, i.e. `CTHeatRecipe`
 - Add remove variants for `HeatRecipe`, targeting `HeatRecipeSimple`, as the others shouldn't really be modifiable via scripts as they are too dynamic.